### PR TITLE
Add a test case about special location URI for CREATE FOREIGN TABLE syntax

### PR DESCRIPTION
**Pull request by jingwen-yang-yjw**
Add a test case that external table location URI contains special character '|' for
CREATE FOREIGN TABLE syntax.

This PR complements test cases of previous PR https://github.com/greenplum-db/gpdb/pull/17199.

### CODE REVIEW

File: gpcontrib/gp_exttable_fdw/output/gp_exttable_fdw.source [hunk 1]
```diff
@@ -71,3 +71,18 @@ SELECT * FROM tableless_ext_fdw;
  3 | 32
 (3 rows)
 
+-- OK, location URI contains special characters '|' and '\'
+-- Single character '|' is used as delimiter for multiple location URIs.
+-- When using CREATE FOREIGN TABLE syntax, '|' and '\' need to be escaped as '|' --> '\|' and '|' --> '\\'.
+CREATE FOREIGN TABLE ext_special_uri(a int, b int)
+SERVER gp_exttable_server
+OPTIONS (format 'csv', delimiter ',',
+         location_uris 'file://localhost.localdomain/home/gpadmin/gpdb_7/gpcontrib/gp_exttable_fdw/data/spe\\cial1\||file://localhost.localdomain/home/gpadmin/gpdb_7/gpcontrib/gp_exttable_fdw/data/\|special2\\');
```
Comment (adam8157) [inline gpcontrib/gp_exttable_fdw/output/gp_exttable_fdw.source:80]: `@hostname@@abs_srcdir@`

**Review by adam8157 — COMMENTED** [review]
Suggest also querying `SELECT urilocation FROM pg_exttable WHERE reloid = 'public.ext_special_uri'::regclass;`
and
`SELECT ftoptions FROM pg_foreign_table WHERE ftrelid='public.multi_uris'::regclass;`

**Comment by jingwen-yang-yjw** [comment]
> Suggest also querying `SELECT urilocation FROM pg_exttable WHERE reloid = 'public.ext_special_uri'::regclass;` and `SELECT ftoptions FROM pg_foreign_table WHERE ftrelid='public.multi_uris'::regclass;`

Done. Thanks for your suggestion!

**Review by hyongtao-db — APPROVED** [review]
LGTM. Thanks.

File: gpcontrib/gp_exttable_fdw/input/gp_exttable_fdw.source [hunk 2]
```diff
@@ -67,3 +67,13 @@ OPTIONS (format 'csv', delimiter ',',
...
+-- OK, location URI contains special characters '|' and '\'
+-- Single character '|' is used as delimiter for multiple location URIs.
+-- When using CREATE FOREIGN TABLE syntax, '|' and '\' need to be escaped as '|' --> '\|' and '|' --> '\\'.
+CREATE FOREIGN TABLE ext_special_uri(a int, b int)
+SERVER gp_exttable_server
+OPTIONS (format 'csv', delimiter ',',
+         location_uris 'file://@hostname@@abs_srcdir@/data/spe\\cial1\||file://@hostname@@abs_srcdir@/data/\|special2\\');
+SELECT urilocation FROM pg_exttable WHERE reloid = 'public.ext_special_uri'::regclass;
+SELECT ftoptions FROM pg_foreign_table WHERE ftrelid='public.ext_special_uri'::regclass;
+SELECT * FROM ext_special_uri ORDER BY a;
```
Comment (hyongtao-db) [inline gpcontrib/gp_exttable_fdw/input/gp_exttable_fdw.source:79]: Pls add a new line after it.

Comment (jingwen-yang-yjw) [reply gpcontrib/gp_exttable_fdw/input/gp_exttable_fdw.source]: Done.

**Review by adam8157 — APPROVED** [review]

**Pull request merged**